app/Http/Controllers/Auth/LoginController.php

```diff
@@ -36,161 +28,78 @@ class LoginController extends BaseController
     }
 
-    public function getLogin()
+    public function login(Request $request)
     {
-        if (Auth::guard('vatsim-sso')->check() && ! Auth::check()) {
-            return $this->attemptSecondaryAuth();
-        } else {
-            return redirect()->route('dashboard');
-        }
-    }
+        if (! $request->has('code') || ! $request->has('state')) {
+            $authorizationUrl = $this->provider->getAuthorizationUrl();
+            $request->session()->put('vatsimauthstate', $this->provider->getState());
 
-    /**
-     * Handle a login request to the application.
-     *
-     * @param  \Illuminate\Http\Request $request
-     * @return \Illuminate\Http\RedirectResponse|\Illuminate\Http\Response
-     */
-    public function loginMain(Request $request)
-    {
-
-        // user has not been authenticated with VATSIM SSO
-        if (! Auth::guard('vatsim-sso')->check()) {
-            return $this->attemptVatsimAuth();
+            return redirect()->away($authorizationUrl);
         }
 
-        if (! Auth::check()) {
-            $this->attemptSecondaryAuth();
+        if ($request->input('state') !== session()->pull('vatsimauthstate')) {
+            return redirect()->route('dashboard')->withError('Something went wrong, please try again.');
         }
 
-        return redirect()->intended(route('mship.manage.dashboard'));
-    }
-
-    protected function getVatsimAuth()
-    {
-        return Auth::guard('vatsim-sso')->user()->id;
+        return $this->verifyLogin($request);
     }
 
-    protected function attemptVatsimAuth()
+    protected function verifyLogin(Request $request)
     {
-        $allowSuspended = true;
-        $allowInactive = true;
-        $token = VatsimSSO::requestToken(route('auth-vatsim-sso'), $allowSuspended, $allowInactive);
-        if ($token) {
-            $key = $token->token->oauth_token;
-            $secret = $token->token->oauth_token_secret;
-            Session::put('credentials.vatsim-sso', compact('key', 'secret'));
-
-            return redirect()->to(VatsimSSO::sendToVatsim());
+        try {
+            $accessToken = $this->provider->getAccessToken('authorization_code', [
+                'code' => $request->input('code'),
+            ]);
+        } catch (IdentityProviderException $e) {
+            return redirect()->route('dashboard')->withError('Something went wrong, please try again.');
         }
-        // Check if there was a CURL error code
-        if (VATSIMSSO::error()['code']) {
-            Log::error('VATSIMSSO was unable to reach CERT. Code:'.VATSIMSSO::error()['code'].' Message:'.VATSIMSSO::error()['message']);
 
-            return redirect()->back()->withErrors(['connection' => "We were unable to contact VATSIM's certification service. Please try again later. If this persists, please contact Web Services."]);
+        $resourceOwner = json_decode(json_encode($this->provider->getResourceOwner($accessToken)->toArray()));
+
+        if (
```

Comment:
Might be worth destructuring the parameters a little to prevent repeat references
```php
[$cid, $personal, $vatsim, $oauth] = $resourceOwner->data
```